includes/providers/gam/class-gam-scripts.php

```diff
@@ -194,12 +228,48 @@ function( $a, $b ) {
 					if( ad_unit['fluid'] ) {
 						baseSizes = baseSizes.concat( 'fluid' );
 					}
+
 					<?php
-					// Iterate through size map.
+					/**
+					 * Identify the bounds container for this slot and use its offset
+					 * width as bounds width.
+					 */
 					?>
+					var boundsWidth = 0;
+					findContainer:
+					for ( var i = 0; i < ad_unit['bounds_selectors'].length; i++ ) {
+						var selector = ad_unit['bounds_selectors'][ i ];
+						if ( typeof boundsContainers[ selector ] === 'undefined' ) {
+							boundsContainers[ selector ] = document.querySelectorAll( selector );
```

Comment:
Non-breaking: why is it necessary to cache the `boundsContainers` for each ad unit? It seems like you're only using it within this loop and never again?

Reply:
This loop is executed for each ad unit on the page because we want to be able to have the `boundsContainers` filterable per ad unit.

Performance-wise, the majority of cases will not be affected by the decision of allowing different values per ad unit because of this caching strategy.